### ISSUE TITLE
Can't seem to reconstruct release-3.3.4

### ISSUE THREAD

**Issue by fernewelten** [adventuregamestudio/ags#810]
Release 3.3.4 is claimed to be the latest stable version on the AGS homepage. I'm trying to build it. I'm working with a Visual Studio 17 with a Visual Studio 15 codetrain.

When checking out the solution `AGS.Editor.Full` of the version, I'm shown the following popup and forced to "upgrade" the solution:
![image](https://user-images.githubusercontent.com/41213694/58772213-29023d00-85b8-11e9-8184-59140ef3d9f1.png)

Subsequent compilation yields more than 200 errors. Evidence suggests that some macros must be incorrectly set because a lot of errors seem to concern standard header files.

For comparison, I check out release 3.4.3 (commit 93f00ce7). Again, Visual Studio 17 forces me to "upgrade" the Editor project. Compiling still yields 41 errors, which is much better than 200 errors but doesn't make me happy.

Typical error message: For file NativeProxy.cs, line 66, 
`Error	CS0012	The type 'Game' is defined in an assembly that is not referenced. You must add a reference to assembly 'AGS.Types, Version=3.99.99.0, Culture=neutral, PublicKeyToken=c617a617b5b1193f'.	AGSEditor	C:\Users\Peter G Bouillon\source\repos\ags\Editor\AGS.Editor\NativeProxy.cs`

Not knowing much c#, I'm not sure whether the "upgrading" has induced the errors or whether the releases simply don't compile out-of-the-box.

**Comment by ivan-mogilko** [comment]
Versions **before 3.5.0** required specifically MSVS 2008 SP1 (professional edition) to compile fully. IIRC one of the reasons was that they used precompiled libraries built with MSVS 2008 which had to be statically linked.
It took us some time to upgrade because not only projects had to be upgraded, but also libraries, and maybe something else.

Could you tell why do you want to compile these versions in particular?

**Comment by fernewelten** [comment]
My main interest is having a reliable release  to run my tests in. I need to be sure that the cases would work when no changes are added so that I can be sure that my changes are indeed at fault when those are added and run into a snafu. 

In other words, I'd like to set up the following workflow:
1. check out and compile the provably reliable version
2. run test for some specific, narrowed-down feature, note results
3. check out and compile version with my added code
4. run same test for same feature, note results
5. compare test run results of both versions.
6. Repair my code if necessary; or else repeat for some different specific feature

I've worked with a very similar workflow with the Googletests so far, but there are things that can't be googletested.

There's nothing enticing in particular in release-3.3.4. I only turned to it when I couldn't get either master latest version or 3.5 latest version to compile and run out-of-the-box.
